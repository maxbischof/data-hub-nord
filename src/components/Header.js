import React from 'react'
import styled from 'styled-components'
import Logo from '../icons/datahubnord_logo.js'
import { useLocation } from 'react-router-dom'

export default function Header() {
  const isRootPath = useLocation().pathname === "/"
  const color = isRootPath ? "white" : "var(--grey)"
  const position = isRootPath ? "absolute" : "static"

  return (
    <StyledHeader color={color} position={position}>
      <a href="/">
        <Logo color={color}/>
        <span>DataHubNord</span>
      </a>
    </StyledHeader>
  )
}

const StyledHeader = styled.header`
  position: ${props => props.position};
  
  span {
    margin: 0 0 0 12px;
  }

  a {
    color: ${props => props.color};
    font-weight: 700;
    padding: 26px 0 30px 22px;
    display: flex;
    align-items: center;
  }

  a:visited {
    color: ${props => props.color};
  }
`

