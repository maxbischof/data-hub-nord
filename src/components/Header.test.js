import React from 'react'
import Header from './Header'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

it('renders datahubnord text', () => {
  const { getByText } = render(
    <MemoryRouter initialEntries={['/']}>
      <Header />
    </MemoryRouter>
  )
  expect(getByText('DataHubNord')).toBeInTheDocument()
})

it('renders header with rootpage style ', () => {
  const { container, getByRole } = render(
    <MemoryRouter initialEntries={['/']}>
      <Header />
    </MemoryRouter>
  )
  const link = getByRole('link')
  expect(link).toHaveStyle('color: white')

  const svg = container.querySelector('svg')
  expect(svg).toHaveStyle('stroke: white')

  const header = container.querySelector('header')
  expect(header).toHaveStyle('position: absolute')
})

it('renders header with datasetdetails style ', () => {
  const { container, getByRole } = render(
    <MemoryRouter initialEntries={[`/datensaetze/Arbeitslosigkeit-Kiel-1`]}>
      <Header />
    </MemoryRouter>
  )
  const link = getByRole('link')
  expect(link).toHaveStyle('color: var(--grey)')

  const svg = container.querySelector('svg')
  expect(svg).toHaveStyle('stroke: var(--grey)')

  const header = container.querySelector('header')
  expect(header).toHaveStyle('position: static')
})
