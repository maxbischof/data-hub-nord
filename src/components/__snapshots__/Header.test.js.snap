// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header component 1`] = `
<header
  className="sc-AxjAm WsNFQ"
>
  <a
    href="/"
  >
    <svg>
      datahubnord_logo.svg
    </svg>
    DataHubNord
  </a>
</header>
`;
